### PR TITLE
Extract some parts of miner from ethcore.

### PR DESCRIPTION
**Pull request by tomusdrw**
Following my new rule of not introducing any new code to `ethcore` crate. New crate is not dependent on `ethcore`, only on newly extracted `ethcore-transaction`.

The changes are required to integrate new Transaction Pool.

Conflicts a bit with #7038 but most of the issues is trivial, will be happy to help @0x7CFE with resolving the conflicts if necessary.

### CODE REVIEW

File: miner/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[package]
+description = "Parity Miner interface."
```
Comment (rphmeier) [inline miner/Cargo.toml]: Why don't we describe it as "block creation for mining or sealing"

**Comment by rphmeier** [comment]
LGTM -- mostly just fixing references to moved code.

**Comment by debris** [comment]
I'll merge master into this branch

**Comment by debris** [comment]
I'll merge it as soon as CI is green

**Pull request merged**